vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt

```diff
@@ -115,7 +116,7 @@ class AndroidDeviceTestRunner(private val device: AndroidDevice) {
             max(configuration.testOutputTimeoutMillis, (testBatch.maxExpectedTestDurationMs * MAX_TEST_DURATION_LEEWAY).toLong())
         )
 
-        // Batch duration timeout can't be less then 'testBatchTimeoutMillis'. There are also additional 10% for unexpected cases.
+        // Batch duration timeout can't be less then 'testBatchTimeoutMillis'. There are also additional 30% for unexpected cases.
```

Comment:
Keep in mind that the timeout is not fully respected by ddmlib: it is only taken into account for some part of the test execution command. If you want to have a reliable timeout management - consider merging with latest develop and try out the adam vendor module. Adam has full transparency in terms of timout management: no concurrent threads executed and locks/latches.